hack/test-go.sh

```diff
@@ -256,18 +256,27 @@ runTests() {
   # must make sure the output from parallel runs is not mixed. To achieve this,
   # we spawn a subshell for each parallel process, redirecting the output to
   # separate files.
-  printf "%s\n" "${@}" | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
-    bash -c "set -o pipefail; _pkg=\"{}\"; _pkg_out=\${_pkg//\//_}; \
-        go test ${goflags[@]:+${goflags[@]}} \
-          ${KUBE_RACE} \
-          ${KUBE_TIMEOUT} \
-          -cover -covermode=\"${KUBE_COVERMODE}\" \
-          -coverprofile=\"${cover_report_dir}/\${_pkg}/${cover_profile}\" \
-          \"${KUBE_GO_PACKAGE}/\${_pkg}\" \
-          ${testargs[@]:+${testargs[@]}} \
-        | tee ${junit_filename_prefix:+\"${junit_filename_prefix}-\$_pkg_out.stdout\"} \
-        | grep \"${go_test_grep_pattern}\"" \
+  if [ "$(uname)" == "Darwin" ]; then
+    export goflags KUBE_RACE KUBE_TIMEOUT KUBE_COVERMODE cover_report_dir \
+    cover_profile testargs junit_filename_prefix go_test_grep_pattern
+
+    echo "${@}" | xargs -n1 -P2 -I{} \
+    bash -c "source hack/lib/golang.sh;kube::golang::run_go_test {}" \
```

Comment:
if we ultimately decide this is the best option, I'd probably prefer to make this in a separate file rather than golang.sh, so you could just do `hack/run-go-test.sh {}` and call that in both the OS X and Linux cases.

we'd want to explicitly note all of the environment variables that must be set in that script, too.